File: arcgis_map/android/src/main/kotlin/de/tollcollect/sv/smartmv/ArcgisMapOptions.kt
package com.example.arcgis_map.type

data class ArcgisMapOptions(
    val apiKey: String,
    val basemap: String,
    val initialCenter: LatLng,
    val isInteractive: Boolean,
    val zoom: Double,
    val hideDefaultZoomButtons: Boolean,
    val hideAttribution: Boolean,
    val padding: ViewPadding,
    val rotationEnabled: Boolean,
    val minZoom: String,
    val maxZoom: Int,
    val xMin: Double,
    val xMax: Double,
    val yMin: Double,
    val yMax: Double,
)

class ViewPadding(
    val left: Double,
    val top: Double,
    val right: Double,
    val bottom: Double,
)

File: arcgis_map_ios/ios/Classes/Models/ArcgisMapOptions.swift
//
//  ArcgisMapOptions.swift
//  arcgis_map
//
//  Created by Julian Bissekkou on 10.10.22.
//

import Foundation

struct ArcgisMapOptions: Codable {
    let apiKey: String
    let basemap: String?
    let vectorTilesUrls: Array<String>?
    let initialCenter: LatLng;
    let zoom: Double
    let padding: ViewPadding
    let isInteractive: Bool
    let rotationEnabled: Bool
    let minZoom: Int
    let maxZoom: Int
    let xMin: Int
    let xMax: Int
    let yMin: Int
    let yMax: Int
}
